subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/SerializedLambdaParametersCheckingCodec.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.serialization.codecs
+
+import org.gradle.configurationcache.problems.DocumentationSection
+import org.gradle.configurationcache.serialization.Codec
+import org.gradle.configurationcache.serialization.IsolateContext
+import org.gradle.configurationcache.serialization.ReadContext
+import org.gradle.configurationcache.serialization.WriteContext
+import org.gradle.configurationcache.serialization.beans.unsupportedFieldDeclaredTypes
+import org.gradle.configurationcache.serialization.logUnsupported
+import org.gradle.configurationcache.serialization.readNonNull
+import org.objectweb.asm.Type
+import org.objectweb.asm.Type.getArgumentTypes
+import java.lang.invoke.SerializedLambda
+import kotlin.reflect.KClass
+
+
+/**
+ * In encoding a [SerializedLambda], this coded checks if the lambda parameter types are allowed in the configuration cache.
+ * If some of them are not, it logs a problem.
+ *
+ * The lambda is encoded straightforwardly as a [BeanSpec], and, upon decoding, a [BeanSpec] is expected to contain the [SerializedLambda].
+ * Beside the compliance checks, the values are encoded or decoded as beans without any special handling.
+ *
+ * @see [org.gradle.configurationcache.serialization.beans.unsupportedFieldDeclaredTypes]
+ */
+object SerializedLambdaParametersCheckingCodec : Codec<SerializedLambda> {
+    override suspend fun ReadContext.decode(): SerializedLambda {
+        return readNonNull<BeanSpec>().bean as SerializedLambda
```

Comment:
🚀 Using `decodeBean()` here would be slightly faster (avoid a recursion roundtrip) and simpler.

Reply:
Done in https://github.com/gradle/gradle/pull/23156/commits/aedb6804f13ee945db7ec1018eac87dbf7b963f9